### PR TITLE
Australia region selector on fronts

### LINKED ISSUE #25292: Australia region selector can get seperated from its sibling container
In Australia we support region specific containers for different parts of Australia, this allows us to personalize the Australia network front based on a users Location.

The region selector is configured as a thrasher which essentially means that Frontend treats it as a seperate container with absolutely no relation to the region specific container, this creates issues as the Labs container can sometimes decide to render between the 2 containers.
image

Whats interesting is that we decide which containers to render server side, not client side. The thrasher used to change region is fairly basic, all it does is set a cookie and then reload the page.

In theory we should be able to get rid of the region selector thrasher and build it natively into the region container itself, but this will need some investigation.

### LINKED ISSUE #25394: Improve accessibility of AU Territory selector
Ensure that buttons for the region selector in Australia are as accessible as possible.

### PR DESCRIPTION
**Pull request by OllysCoding**
## What does this change?

Adds a 'region selector' to any facia container targeted for regions in Australia (Victoria, New South Wales, Queensland) which allows a user to switch into different regions (e.g if Fastly geo-location got it wrong).

This is currently a thrasher - where sometimes it is separated from the intended container by the high merchandising slot, which is why we're moving it into Frontend. More info: https://github.com/guardian/frontend/issues/25292

This PR adds this feature behind a feature switch `aus-region-selector`, meaning we can safely merge this PR without rolling it out.

> ⚠️ Before this feature is switched on, the thrashers should be _removed_ to avoid duplicate UI

As part of moving this across from a thrasher, we've tried to do this the 'Frontend way' wherever possible, using `@guardian/libs` for cookie handling & Frontend font handling (meaning there are slight variations in how text looks)

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes - This change will need to be migrated to DCR at some point in the future

## Screenshots

| Before (Thrasher)      | After  (Frontend)    |
|-------------|------------|
| ![before][] | ![after][] |
| ![before1][] | ![after1][] |

[before]: https://user-images.githubusercontent.com/9575458/183881563-ea0e8cb2-9254-48a8-a76b-9e79f0d5a233.png
[after]: https://user-images.githubusercontent.com/9575458/183881644-e44c21ca-33e8-4a9b-bdf1-39aab5b03532.png
[before1]: https://user-images.githubusercontent.com/9575458/183881798-fbd13386-ce87-4a57-967b-e9be1860eaad.png
[after1]: https://user-images.githubusercontent.com/9575458/183881858-13fab14b-f822-4c2c-aea3-6899779bfde8.png

### Accessibility test checklist

- [x] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [x] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Comment by HarryFischer** [comment]
@OllysCoding Looks good - 1 small thing can we get the button text in textSans please rather than serif

File: common/app/conf/switches/FeatureSwitches.scala [hunk 1]
```diff
@@ -492,4 +492,14 @@ trait FeatureSwitches {
     sellByDate = never,
     exposeClientSide = true,
   )
+
+  val AusRegionSelector = Switch(
+    SwitchGroup.Feature,
+    "aus-region-seelctor",
```
Comment (lucymonie) [inline common/app/conf/switches/FeatureSwitches.scala:498]:
```suggestion
    "aus-region-selector",
```

File: common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
+@import views.support.RenderClasses
+@(containerDefinition: layout.FaciaContainer)(implicit request: RequestHeader)
+
+<div class='fc-aus-territory__container'>
+    <button
+    class='fc-aus-territory__accordion-header'
+    data-link-name="AU-NSW-expand"
```
Comment (lucymonie) [inline common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html:7]: I may be misunderstanding this (apologies if so) but should this data-link-name value be dynamic depending on the territory of the container? Presumably the 'NSW' bit could be replaced with  some mutation of `@containerDefinition.territoryName`?

File: common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        <button class="@RenderClasses(Map(
+                ("u-button-reset", true),
+                ("fc-aus-territory__territory-button", true),
+                ("fc-aus-territory__territory-button--selected", containerDefinition.targetedTerritory.exists(_.id == "AU-VIC")),
+            ))"
+            data-link-name="AU-VIC">
+            <span class=''>Victoria</span>
+        </button>
+        <button class='u-button-reset fc-aus-territory__territory-button fc-aus-territory__territory-button--secondary'
+            data-link-name="AU-other"
```
Comment (lucymonie) [inline common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html:72]: For consistency of data collection, this was `data-link-name="AU-hide-thrasher"` but if you have a good reason to change the value, please ignore me!

File: static/src/javascripts/projects/facia/modules/ui/au-region-selector.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+			container,
+		);
+		territoryButtons.each((button) => {
+			button.addEventListener('click', () => {
+				const attrValue = button.getAttribute('data-link-name');
+				if (attrValue) {
+					// Fastly expects the value of 'Other' - if a different, non-valid value is provided,
+					// fastly will ignore the cookie & use the users locations
+					const value =
+						attrValue === 'AU-other' ? 'Other' : attrValue;
```
Comment (lucymonie) [inline static/src/javascripts/projects/facia/modules/ui/au-region-selector.js:34]: Goodo - this is the main place where you could go wrong, and you didn't 😄

**Review by lucymonie — APPROVED** [review]
LGTM 👍🏻

File: static/src/javascripts/projects/facia/modules/ui/au-region-selector.js (hunk 4 above)
Comment (OllysCoding) [reply static/src/javascripts/projects/facia/modules/ui/au-region-selector.js:37]: :D

File: common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html (hunk 3 above)
Comment (OllysCoding) [reply common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html:79]: Will change it back!

File: common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html (hunk 2 above)
Comment (OllysCoding) [reply common/app/views/fragments/containers/facia_cards/auTerritorySelector.scala.html]: Good catch!

**Review by mxdvl — APPROVED** [review]
It would have been nice to [have some type safety in the UI script](https://github.com/guardian/frontend/pull/25361/files#diff-bba4e6a1198971c29eb8f41bc454c464e9837c139a08788ae64e9b3e334960a3), but I undertand that this will soon be replaced by DCR anyways.

EDIT: I raised PR #25400 as a propsal

**Comment by OllysCoding** [comment]
> I raised PR https://github.com/guardian/frontend/pull/25400 as a propsal
This is great thankyou! I'd actually originally created the file as .ts but didn't manage to figure out there were TS variants for the UI libs & gave up on typing it - will look at this PR now!

**Pull request merged**